File: src/main/java/com/dongtronic/diabot/commands/RolesCommand.java
package com.dongtronic.diabot.commands;

import com.jagrosh.jdautilities.command.Command;
import com.jagrosh.jdautilities.command.CommandEvent;
import com.jagrosh.jdautilities.doc.standard.CommandInfo;
import net.dv8tion.jda.core.Permission;
import net.dv8tion.jda.core.entities.Guild;
import net.dv8tion.jda.core.entities.Role;

import java.util.ArrayList;
import java.util.List;

@CommandInfo(
    name = {"Roles"},
    description = "Get all roles in the server"
)
public class RolesCommand extends Command {

  public RolesCommand(Category category) {
    this.name = "roles";
    this.help = "Get all roles in the server";
    this.guildOnly = true;
    this.category = category;
    this.userPermissions = new Permission[]{Permission.ADMINISTRATOR};
  }

  @Override
  protected void execute(CommandEvent event) {

    Guild guild = event.getGuild();

    List<Role> roles = guild.getRoles();

    StringBuilder returned = new StringBuilder().append("```");

    for(Role role : roles) {
      returned.append("\n" + role.getId() + " - " + role.getName());
    }

    returned.append("\n```");

    event.reply(returned.toString());
  }
}

File: src/main/java/com/dongtronic/diabot/commands/RolesCommand.kt
package com.dongtronic.diabot.commands

import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent
import net.dv8tion.jda.core.EmbedBuilder
import net.dv8tion.jda.core.Permission

class RolesCommand(category: Command.Category) : DiabotCommand() {

    init {
        this.name = "roles"
        this.help = "Get all roles in the server"
        this.guildOnly = true
        this.category = category
        this.userPermissions = arrayOf(Permission.ADMINISTRATOR)
    }

    override fun execute(event: CommandEvent) {

        val guild = event.guild
        val roles = guild.roles

        val builder = EmbedBuilder()

        builder.setTitle("Roles for ${event.guild.name}")

        for (role in roles) {
            builder.addField(role.name, role.id, true)
        }

        event.reply(builder.build())
    }
}
